blake2/src/blake2.rs

```diff
@@ -279,56 +281,61 @@ macro_rules! blake2_impl {
 
         #[derive(Clone)]
         #[doc=$doc]
-        pub struct $fix_state {
+        pub struct $fix_state<U = $bytes> {
             state: $state,
+            _u: PhantomData<U>,
         }
 
-        impl Default for $fix_state {
+        impl<U: Unsigned> Default for $fix_state<U> {
             fn default() -> Self {
-                let state = $state::new_keyed(&[], $bytes::to_usize());
-                Self { state }
+                let state = $state::new_keyed(&[], U::to_usize());
+                Self { state, _u: PhantomData }
             }
         }
 
-        impl BlockInput for $fix_state {
+        impl<U> BlockInput for $fix_state<U> {
             type BlockSize = $bytes;
```

Comment:
It seems (from wikipedia) that block size should be 32 for Blake2b256. But what about intermediate values?

Reply:
Changing output size does not change the block size.